tozny/resource_realm_group.go

```diff
@@ -76,10 +77,132 @@ func resourceRealmGroup() *schema.Resource {
+							Type:        schema.TypeInt,
+							Optional:    true,
+						},
+						"plugin_type": {
+							Description: "The supported plugin type for the access policy (e.g. \"jira\").",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"plugin_id": {
+							Description: "The ID of the plugin.",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"plugin_mpc_flow_source": {
+							Description: "The ID of the source that managed MPC (e.g. the ID of a Jira Board)",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
+// Converts the access policies resource from Terraform to a list of AccessPolicy
+func accessPoliciesFromTerraform(data []interface{}) []identityClient.AccessPolicy {
+	var policies []identityClient.AccessPolicy
+	for _, terraformPolicy := range data {
+		policy := accessPolicyFromTerraform(terraformPolicy.(map[string]interface{}))
+		policies = append(policies, policy)
+	}
+	return policies
+}
+
+// Converts a single access policy argument from Terraform to the AccessPolicy API
+func accessPolicyFromTerraform(data map[string]interface{}) identityClient.AccessPolicy {
+	approvalRoles := approvalRolesFromTerraform(data["approval_role_ids"].([]interface{}))
+	policy := identityClient.AccessPolicy{
+		ID:                           int64(data["access_policy_id"].(int)),
+		ApprovalRoles:                approvalRoles,
+		RequiredApprovals:            data["required_approvals"].(int),
+		MaximumAccessDurationSeconds: data["maximum_access_duration_seconds"].(int),
+		PluginType:                   data["plugin_type"].(string),
+		PluginID:                     data["plugin_id"].(string),
+		PluginMPCFlowSource:          data["plugin_mpc_flow_source"].(string),
+	}
+	return policy
+}
+
+// Converts the approval roles argument from Terraform to a list of Roles
+func approvalRolesFromTerraform(data []interface{}) []identityClient.Role {
+	var roles []identityClient.Role
+	for _, terraformApprovalRoleID := range data {
+		role := identityClient.Role{
+			ID: terraformApprovalRoleID.(string),
+		}
+		roles = append(roles, role)
+	}
+	return roles
+}
+
+// Converts a list of Role IDs to a list of Terraform approval role ids
+func approvalRoleIdsToTerraform(roles []identityClient.Role) []interface{} {
+	var ids []interface{}
+	for _, role := range roles {
+		ids = append(ids, role.ID)
+	}
+	return ids
+}
+
+// Iterates through the API's access policies and maps the values to a structure that Terraform can assign
+// to the Group resource's access_policies argument
+func flattenAccessPolicyItems(groupAccessPolicy identityClient.GroupAccessPolicies) []interface{} {
+	// If no access policies, return empty interface
+	if len(groupAccessPolicy.AccessPolicies) == 0 {
+		return make([]interface{}, 0)
+	}
+
+	// Build each flattened policy and append to aps
```

Comment:
aps == access policy

Reply:
aps == access policies 
ap == access policy 

I will employ better naming conventions

Reply:
❤️

Reply:
code is read more than its written, better to optimize for the future readers (including yourself)